src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -940,6 +940,9 @@ internal void CompleteOnCurrentThread()
                             }
                             else if (tryAgainAfter is object)
                             {
+                                // prevent referencing tasks which may be GCed during the waiting cycle.
+                                visited?.Clear();
```

Comment:
not directly related, but this table retains tasks otherwise can be GCed. It leads lots of problems to write tests related to GCed tasks, which turns out not necessary to reproduce this problem.